packages/lexicon/src/validators/complex.ts

```diff
@@ -154,6 +157,24 @@ export function object(
   return { success: true, value: resultValue }
 }
 
+export function record(
+  lexicons: Lexicons,
+  path: string,
+  def: LexRecord,
+  value: unknown,
+): ValidationResult {
+  if (!isObj(value) || !('$type' in value)) {
```

Comment:
We can use this helper:
```suggestion
  if (!isDiscriminatedObject(value)) {
```